tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h

```diff
@@ -102,8 +109,26 @@ struct OptimizationProfileConfig {
     }
     return true;
   }
+
+ private:
+  // In dynamic shape mode the optimization profile dims are only allowed to
+  // differ from the network input dims where the network input dims have -1
+  // values. We ensure that prof_dim matches the network input (mask).
+  void applyMask(nvinfer1::Dims* prof_dim, const nvinfer1::Dims& mask) {
+    for (int i = 0; i < mask.nbDims; i++) {
+      if (mask.d[i] != -1) {
+        prof_dim->d[i] = mask.d[i];
```

Comment:
More question on this: why we overwrite the existing value in the prof_dim? We need to add a comment here if that is what we have to do. Should we accumulate instead?
By accumulate, here is what is mean:
     if prof_dim is min, then new prof_dim  = min(old prof_dim, mask.d)
     if prof_dim is max, the new prof_dim = max(old prof_dim, mask.d)
     no idea on what to do if prof_dim is opt though

Reply:
There is no accumulation here, any accumulation is already done by `TrtShapeOptimizationProfile::InitProfiles()`

Please have another look in light of the updated docstring of SetDimensions.